lib/msf/util/host.rb

```diff
@@ -52,7 +49,6 @@ def self.normalize_host(host)
         # to try, just log it and return what we were given
         if !norm_host
           dlog("Host could not be normalized: #{host.inspect}")
-          norm_host = host
```

Comment:
Hm. It looks like some code paths are expecting the original value to be returned still for validation purposes:
https://github.com/rapid7/metasploit-framework/blob/c0185f65bfb7cb6d280cab7c47a34310d0e2e390/lib/msf/core/db_manager/host.rb#L199-L203

Which I believe is called by report_note:

https://github.com/rapid7/metasploit-framework/blob/c0185f65bfb7cb6d280cab7c47a34310d0e2e390/lib/msf/core/db_manager/note.rb#L94-L95

I wonder if the enum_dns module needs to explicitly attempt resolve the host itself first, then provide nil to report_note if things aren't resolvable 🤔